### DISCUSSION TITLE
Is it possible run unit tests when we save a file?

### DISCUSSION THREAD

**Discussion by peczenyj** [zed-industries/zed#15003] [Feature Requests]
Hello

I am a golang dev trying to use Zed Editor

I note that is possible run a formatter when we save a file, but is it possible run the unit tests when we save a file?

VSCode can be configured to run the golang unit tests on the same folder when we save a file, and this provides me a quick feedback of what I am doing instead have to type `make test` or something else

**Comment by omarjatoi** [comment]
I think the preferred way to run tests would be using "Tasks" (docs: https://zed.dev/docs/tasks). Hypothetically, you _could_ hack together a command to run instead of the formatter (see https://zed.dev/docs/configuring-zed#formatter) so your command could be `make test` which runs on save.

I'd probably encourage you to use tasks 🙂 Hope this helps!

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
